sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -512,6 +527,12 @@ private[hive] trait HiveInspectors {
         _ => constant
       case poi: VoidObjectInspector =>
         _ => null // always be null for void object inspector
+      case ym: WritableConstantHiveIntervalDayTimeObjectInspector =>
+        val constant = ym.getWritableConstantValue.asInstanceOf[HiveIntervalDayTime]
+        _ => constant.getTotalSeconds * DateTimeConstants.NANOS_PER_MICROS + constant.getNanos
```

Comment:
I didn't get this. You multiply seconds by 1000 which is milliseconds and add nanoseconds. Which is obviously wrong (btw day-time interval has microsecond precision). Could you fix this and write a test for this piece of code, please.

Reply:
> I didn't get this. You multiply seconds by 1000 which is milliseconds and add nanoseconds. Which is obviously wrong (btw day-time interval has microsecond precision). Could you fix this and write a test for this piece of code, please.

Ohhh, sorry, I want to write `DateTimeConstants.NANOS_PER_SECOND`